packages/starlight/user-components/Badge.astro

```diff
@@ -0,0 +1,153 @@
+---
+import { BadgeComponentSchema } from '../schemas/badge';
+import { parseWithFriendlyErrors } from '../utils/error-map';
+import type { HTMLAttributes } from 'astro/types';
+
+type Props = {
+	variant?: 'note' | 'danger' | 'success' | 'caution' | 'tip' | 'default';
+	text: string;
+	size?: 'small' | 'medium' | 'large';
+	class?: string | undefined;
+} & HTMLAttributes<'span'>;
+
+const {
+	variant: variantProp,
+	text: textProp,
+	size: sizeProp,
+	class: classProp,
+	...attrs
+} = Astro.props;
+
+const { text, variant, size, class: customClass } = parseWithFriendlyErrors(
+	BadgeComponentSchema,
+	{
+		class: classProp,
+		variant: variantProp,
+		text: textProp,
+		size: sizeProp,
+	},
+	'Invalid prop passed to the `<Badge/>` component.'
+);
```

Comment:
```suggestion
type Props = BadgeComponentProps & HTMLAttributes<'span'>;

const {
	text,
	variant,
	size,
	class: customClass,
	...attrs
} = parseWithFriendlyErrors(
	BadgeComponentSchema,
	Astro.props,
	'Invalid prop passed to the `<Badge/>` component.'
);
```

The previous change allows for a bit of simplification in the component.

Reply:
Thanks Hideo 💜